src/pubsub/ua_pubsub_ns0.c

```diff
@@ -1079,6 +1085,145 @@ publishedDataItemsTypeDestructor(UA_Server *server,
+/*         PubSub configurator       */
+/*************************************/
+
+/* UA_loadPubSubConfigMethodCallback() */
+/**
+ * @brief   callback function that will be executed when the method "PubSub configurator (replace config)" is called.
+ */
+#ifdef UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS 
+#ifdef UA_ENABLE_PUBSUB_CONFIG
+static UA_StatusCode
+UA_loadPubSubConfigMethodCallback(UA_Server *server,
+                                  const UA_NodeId *sessionId, void *sessionHandle,
+                                  const UA_NodeId *methodId, void *methodContext,
+                                  const UA_NodeId *objectId, void *objectContext,
+                                  size_t inputSize, const UA_Variant *input,
+                                  size_t outputSize, UA_Variant *output)
+{
+    UA_StatusCode retVal = UA_STATUSCODE_GOOD;
+    UA_ExtensionObject decodedBinFile;
+    memset(&decodedBinFile, 0, sizeof(UA_ExtensionObject));
+
+    do {
+        UA_ByteString *inputStr = (UA_ByteString*)input->data;
+        retVal = UA_decodeBinFile(inputStr, 0, &decodedBinFile);
+        if(retVal != UA_STATUSCODE_GOOD) {
+            break;
+        }
+
+        UA_PubSubConfigurationDataType *pubSubConfig;
+        retVal = UA_getPubSubConfig(&decodedBinFile, &pubSubConfig);
+        if(retVal != UA_STATUSCODE_GOOD) {
+            break;
+        }
+
+        retVal = UA_updatePubSubConfig(server, pubSubConfig);
+        if(retVal != UA_STATUSCODE_GOOD) {
+            break;
+        }
+    } while(0);
+
+    UA_ExtensionObject_deleteMembers(&decodedBinFile);
+    return retVal;
+}
+#endif
+#endif /*UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS*/
+
+
+/* UA_addLoadPubSubConfigMethod() */
+/**
+ * @brief       Adds method node to server. This method is used to load binary files for PubSub 
+ *              configuration and delete / replace old PubSub configurations.
+ * 
+ * @param       server      [bi]    UA_Server object that shall contain the method.
+ * 
+ * @return      UA_STATUSCODE_GOOD on success
+ */
+#ifdef UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS
+#ifdef UA_ENABLE_PUBSUB_CONFIG
+static UA_StatusCode 
+UA_addLoadPubSubConfigMethod(UA_Server *server)
+{
+    UA_Argument inputArgument;
+    UA_Argument_init(&inputArgument);
+    inputArgument.description = UA_LOCALIZEDTEXT("en-US", "PubSub config binfile");
+    inputArgument.name = UA_STRING("BinFile");
+    inputArgument.dataType = UA_TYPES[UA_TYPES_BYTESTRING].typeId;
+    inputArgument.valueRank = UA_VALUERANK_SCALAR;
+
+    UA_MethodAttributes configAttr = UA_MethodAttributes_default;
+    configAttr.description = UA_LOCALIZEDTEXT("en-US","Load binary configuration file");
+    configAttr.displayName = UA_LOCALIZEDTEXT("en-US","LoadPubSubConfiguration");
+    configAttr.executable = true;
+    configAttr.userExecutable = true;
+    UA_StatusCode retVal = UA_Server_addMethodNode(server, UA_NODEID_STRING(1, "PubSub_replace_config_method"),
```

Comment:
Static string node ids are uncommon. I think it would be better to autogenerate a numeric node ID.